lncrawl/sources/1stkissnovel.py

```diff
@@ -1,81 +1,104 @@
         self.novel_title = possible_title.text.strip()
-        logger.info('Novel title: %s', self.novel_title)
+        logger.info("Novel title: %s", self.novel_title)
 
-        self.novel_cover = soup.select_one(
-            'meta[property="og:image"]')['content']
-        logger.info('Novel cover: %s', self.novel_cover)
+        self.novel_cover = self.absolute_url(
+            soup.select_one(".summary_image a img")["src"]
+        )
+        logger.info("Novel cover: %s", self.novel_cover)
 
-        self.novel_author = ' '.join([
+        self.novel_author = " ".join(
+            [
             a.text.strip()
             for a in soup.select('.author-content a[href*="manga-author"]')
-        ])
-        logger.info('%s', self.novel_author)
+            ]
+        )
+        logger.info("%s", self.novel_author)
+
+        self.novel_id = soup.select_one("#manga-chapters-holder")["data-id"]
+        logger.info("Novel id: %s", self.novel_id)
 
-        volumes = set()
-        chapters = soup.select('ul.main li.wp-manga-chapter a')
-        for a in reversed(chapters):
+        ## For getting cookies
+        #self.submit_form(wp_admin_ajax_url, data={
+        #    'action': 'manga_views',
+        #    'manga': self.novel_id,
+        #})
+        #print(self.cookies)
+        response = self.submit_form(wp_admin_ajax_url, data={
+            'action': 'manga_get_chapters',
+            'manga': self.novel_id,
+        })
+        soup = self.make_soup(response)
+        for a in reversed(soup.select(".wp-manga-chapter a")):
             chap_id = len(self.chapters) + 1
-            vol_id = (chap_id - 1) // 100 + 1
-            volumes.add(vol_id)
-            self.chapters.append({
-                'id': chap_id,
-                'volume': vol_id,
-                'url':  self.absolute_url(a['href']),
-                'title': a.text.strip() or ('Chapter %d' % chap_id),
-            })
+            vol_id = 1 + len(self.chapters) // 100
+            if chap_id % 100 == 1:
+                self.volumes.append({"id": vol_id})
+            # end if
+            self.chapters.append(
+                {
+                    "id": chap_id,
+                    "volume": vol_id,
+                    "title": a.text.strip(),
+                    "url": self.absolute_url(a["href"]),
+                }
+            )
         # end for
 
-        self.volumes = [{'id': x} for x in volumes]
     # end def
 
     def download_chapter_body(self, chapter):
-        '''Download body of a single chapter and return as clean html format.'''
-        logger.info('Downloading %s', chapter['url'])
-        soup = self.get_soup(chapter['url'])
+        """Download body of a single chapter and return as clean html format."""
+        logger.info("Visiting %s", chapter["url"])
+        soup = self.get_soup(chapter["url"])
 
-        contents = soup.select_one('div.text-left')
+        contents = soup.select_one('div.text-left div:first-child')
```

Comment:
this does not work on many chapters. e.g.: 

- https://1stkissnovel.love/novel/the-reborn-wife-is-too-seductive/chapter-9/
- https://1stkissnovel.love/novel/the-reborn-wife-is-too-seductive/chapter-10/

Reply:
Good catch!